pkg/server/server.go

```diff
@@ -416,8 +416,27 @@ func (s *Server) Run(ctx context.Context) error {
 		s.kcpSharedInformerFactory.Start(ctx.StopCh)
 		s.rootKcpSharedInformerFactory.Start(ctx.StopCh)
 
-		s.kcpSharedInformerFactory.WaitForCacheSync(ctx.StopCh)
-		s.rootKcpSharedInformerFactory.WaitForCacheSync(ctx.StopCh)
+		if err := wait.PollImmediateInfiniteWithContext(goContext(ctx), time.Millisecond*500, func(ctx context.Context) (bool, error) {
```

Comment:
why not start the informers and then wait until they sync in a post-startup hook?

Reply:
this is about the context closing. Was falling a couple of time seeing the post-start-hook continue as if the informers had synced.